src/psyclone/domain/lfric/transformations/lfric_extract_trans.py

```diff
@@ -142,42 +140,13 @@ def apply(self, nodes, options=None):
             # changing the user's options:
             my_options = options.copy()
 
-        ctu = CallTreeUtils()
         nodes = self.get_node_list(nodes)
-        region_name = self.get_unique_region_name(nodes, my_options)
-        my_options["region_name"] = region_name
-        my_options["prefix"] = my_options.get("prefix", "extract")
-        # Get the input- and output-parameters of the node list
-        read_write_info = \
-            ctu.get_in_out_parameters(nodes, collect_non_local_symbols=True)
-
-        # Even variables that are output-only need to be written with their
-        # values at the time the kernel is called: many kernels will only
-        # write to part of a field (e.g. in case of MPI the halo region
-        # will not be written). Since the comparison in the driver uses
-        # the whole field (including values not updated), we need to write
-        # the current value of an output-only field as well. This is
-        # achieved by adding any written-only field to the list of fields
-        # read. This will trigger to write the values in the extraction,
-        # and the driver code created will read in their values.
-        for sig in read_write_info.write_list:
-            if sig not in read_write_info.read_list:
-                read_write_info.read_list.append(sig)
-
-        # Determine a unique postfix to be used for output variables
-        # that avoid any name clashes
-        postfix = ExtractTrans.determine_postfix(read_write_info,
-                                                 postfix="_post")
-        my_options["post_var_postfix"] = postfix
+        super().apply(nodes, my_options)
+        new_node = nodes[0].ancestor(ExtractNode)
         if my_options.get("create_driver", False):
-            # We need to create the driver before inserting the ExtractNode
-            # (since some of the visitors used in driver creation do not
-            # handle an ExtractNode in the tree)
-            self._driver_creator.write_driver(nodes, read_write_info,
-                                              postfix=postfix,
-                                              prefix=my_options["prefix"],
-                                              region_name=region_name)
+            new_node._driver_creator = LFRicExtractDriverCreator()
+        return
+
         # The PSyData transformation needs to pass this object to
```

Comment:
However, deleting the "apply" lost the `param bool options["create_driver"]` and "reads of output-only variables" because this is implemented here but not in the lowering. So I introduced it again and I capture the _dirver_creator in the node (to run it al lowering) instead of here.

This required all the remaining changes in this PR, as I had to move Transformation functionality (determine_postfix, get_unique_region_name) to the Extract/psy_data node, because it is where it is used now.

Reply:
I am happy to remove the create-driver setting, it is now stable enough to be always on (since it doesn't make too much sense to extract data without having a matching driver :) ).

Reply:
What about the "reads of output-only variables", can it be also part of the base Extract class, right now it is done for LFRic and GOcean, but not for generic code extraction?

Reply:
Yes. I have a work-in-progress branch that combines all of this properly into a base class, but I guess with your changes coming in, I need to start from scratch.